presto-main/src/test/java/com/facebook/presto/operator/aggregation/TestApproximatePercentileAggregation.java

```diff
@@ -320,125 +361,206 @@ public void testFloatPartialStep()
+    private void testAggregationDoubles(InternalAggregationFunction function, Page page, double error, List<Double> quantiles, double... inputs)
+    {
+        // test scalars
+        List<Double> rows = Arrays.stream(inputs).sorted().boxed().collect(Collectors.toList());
+        List<Double> returned;
+        try {
+            Double value = (Double) AggregationTestUtils.aggregation(function, page);
+            if (value == null) {
+                returned = new ArrayList<>();
+            }
+            else {
+                returned = ImmutableList.of(value);
+            }
+        }
+        catch (ClassCastException e) {
+            returned = (List<Double>) AggregationTestUtils.aggregation(function, page);
+        }
+        assertPercentilesWithinError(returned, error, rows, quantiles);
+    }
 
-        assertAggregation(
-                DOUBLE_APPROXIMATE_PERCENTILE_WEIGHTED_AGGREGATION,
-                3.0,
-                createDoublesBlock(1.0, null, 2.0, null, 2.0, null, 2.0, null, 3.0, null, 3.0, null, 3.0, 4.0, 5.0, 6.0, 7.0),
-                createLongsBlock(1L, 1L, 2L, 1L, 2L, 1L, 2L, 1L, 2L, 1L, 1L, 1L, 1L, 1L, 1L, 1L, 1L),
-                createRLEBlock(0.5, 17));
+    private void assertPercentilesWithinError(List<Double> result, double error, List<? extends Number> rows, List<Double> percentiles)
+    {
+        if (rows.isEmpty()) {
+            // Nothing to assert except that the tdigest is empty
+            return;
+        }
 
-        assertAggregation(
-                DOUBLE_APPROXIMATE_PERCENTILE_WEIGHTED_AGGREGATION_WITH_ACCURACY,
-                9900.0,
-                createDoubleSequenceBlock(0, 10000),
-                createLongRepeatBlock(1, 10000),
-                createRLEBlock(0.99, 10000),
-                createRLEBlock(0.001, 10000));
+        // Test each quantile individually (value_at_quantile)
+        for (int i = 0; i < result.size(); i++) {
+            assertPercentileWithinError(result.get(i), error, rows, percentiles.get(i));
+        }
+    }
 
-        // weighted + array of approx_percentile
-        assertAggregation(
-                DOUBLE_APPROXIMATE_PERCENTILE_ARRAY_WEIGHTED_AGGREGATION,
-                ImmutableList.of(1.0, 2.0),
-                createDoublesBlock(1.0, 2.0, 3.0),
-                createLongsBlock(4L, 2L, 1L),
-                createRLEBlock(ImmutableList.of(0.5, 0.8), 3));
+    private void assertPercentileWithinError(Double result, double error, List<? extends Number> rows, double percentile)
+    {
+        double lowerBound = Math.max(0, percentile - error);
+        double upperBound = Math.min(1, percentile + error);
+
+        double min = rows.get((int) (rows.size() * lowerBound)).doubleValue();
+        double max = rows.get(min(rows.size() - 1, (int) (rows.size() * upperBound))).doubleValue();
+        // Check that the chosen quantile is within the upper and lower bound of the error
+        functionAssertions.assertFunction(
```

Comment:
It seems like there's just no reason to put the comparison as a SQL function.  Why do this?